clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -845,6 +882,30 @@ private void handleListOffsetResponse(Map<TopicPartition, Long> timestampsToSear
         return parsedRecords;
     }
 
+    private PriorityQueue<FetchResponse.AbortedTransaction> getAbortedTransactions(FetchResponse.PartitionData partition) {
+        PriorityQueue<FetchResponse.AbortedTransaction> abortedTransactions = null;
+        if (partition.abortedTransactions != null && 0 < partition.abortedTransactions.size()) {
+            abortedTransactions = new PriorityQueue<>(
+                    partition.abortedTransactions.size(),
+                    new Comparator<FetchResponse.AbortedTransaction>() {
+                        @Override
+                        public int compare(FetchResponse.AbortedTransaction o1, FetchResponse.AbortedTransaction o2) {
```

Comment:
Hmm, I thought the AbortedTransactions in the FetchResponse is guaranteed to be sorted by the firstOffset on the broker side already?